firefox_puppeteer/tests/test_toolbars.py

```diff
@@ -137,8 +136,9 @@ def test_matching_text(self):
             url_matches = autocompleteresults.get_matching_text(result, "url")
             all_matches = title_matches + url_matches
             self.assertTrue(len(all_matches) > 0)
-            for match_fragment in all_matches:
-                self.assertIn(match_fragment, (input_text, input_text.upper()))
+            # check all but the last item (the new unified search item from Bug 1168811)
+            for match_fragment in all_matches[:-1]:
+                self.assertIn(input_text, match_fragment.lower())
```

Comment:
We do not know how many search items will be visible at the end. So my proposal here would be to limit the auto-completion to history items only. Similar to what we do in the actual tests. That's a safer method.